backend/fms_core/services/sample_next_step.py

```diff
@@ -48,4 +48,125 @@ def queue_sample_to_study_workflow(sample_obj: Sample, study_obj: Study, order:
                                                              study=study_obj)
         except Exception as err:
             errors.append(err)
-    return sample_next_step, errors, warnings
\ No newline at end of file
+    return sample_next_step, errors, warnings
+
+def dequeue_sample_from_specifc_step_study_workflow(sample_obj: Sample, study_obj: Study, order: int) -> Tuple[Union[int, None], List[str], List[str]]:
+    """
+    Deletes a SampleNextStep instance to indicate the removal of a sample from a workflow at a specific step. 
+    Args:
+        `sample_obj`: Sample instance to be dequeued from the study workflow.
+        `study_obj`: Study instance currently associated to the sample.
+        `order`: Positive integer indicating the current position of the sample in the workflow.
+                 
+    Returns:
+        Tuple containing the number of SampleNextStep instances deleted, the error messages and the warning messages. 
+
+    """
+    dequeued = None
+    errors = []
+    warnings = []
+
+    if not isinstance(sample_obj, Sample):
+        sample_obj = None
+        errors.append(f"A valid sample instance must be provided.")
+
+    if not isinstance(study_obj, Study):
+        study_obj = None
+        errors.append(f"A valid study instance must be provided.")
+    
+    if not order:
+        order = None
+        errors.append(f"A step order must be provided.  ")
+    
+    step_order=None
+    # Dequeuing from a specific step
+    if order is not None:
+        try:
+            step_order = StepOrder.objects.get(order=order, workflow=study_obj.workflow)
+        except StepOrder.DoesNotExist:
+            errors.append(f"No step found for the given order.")
+    
+    # Dequeuing from study workflow by deleting the SampleNextStep instance
+    if sample_obj and study_obj and step_order and not errors:
+        try:
+            num_deleted, _ = SampleNextStep.objects.filter(sample=sample_obj, study=study_obj, step_order=step_order).delete()
```

Comment:
This uses the manager delete... use first() to get the model delete and ensure only a single object is deleted (the unique constraint does prevent this for all step_order different from None, but cannot enforce unique on None values).